hazelcast/src/main/java/com/hazelcast/monitor/impl/EmptyLocalReplicatedMapStats.java

```diff
@@ -0,0 +1,252 @@
+        return 0;
+    }
+
+    @Override
+    public long getCreationTime() {
+        return 0;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return 0;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return 0;
+    }
+
+    @Override
+    public long getHits() {
+        return 0;
+    }
+
+    @Override
+    public long getLockedEntryCount() {
+        return 0;
+    }
+
+    @Override
+    public long getDirtyEntryCount() {
+        return 0;
+    }
+
+    @Override
+    public long getPutOperationCount() {
+        return 0;
+    }
+
+    @Override
+    public long getGetOperationCount() {
+        return 0;
+    }
+
+    @Override
+    public long getRemoveOperationCount() {
+        return 0;
+    }
+
+    @Override
+    public long getTotalPutLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getTotalGetLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getTotalRemoveLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getMaxPutLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getMaxGetLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getMaxRemoveLatency() {
+        return 0;
+    }
+
+    @Override
+    public long getEventOperationCount() {
+        return 0;
+    }
+
+    @Override
+    public long getOtherOperationCount() {
+        return 0;
+    }
+
+    @Override
+    public long total() {
+        return 0;
+    }
+
+    @Override
+    public long getHeapCost() {
+        return 0;
+    }
+
+    @Override
+    public long getMerkleTreesCost() {
+        return 0;
+    }
+
+    @Override
+    public NearCacheStats getNearCacheStats() {
+        throw new UnsupportedOperationException("Replicated map has no Near Cache!");
+    }
+
+    @Override
+    public long getQueryCount() {
+        throw new UnsupportedOperationException("Queries on replicated maps are not supported.");
+    }
+
+    @Override
+    public long getIndexedQueryCount() {
+        throw new UnsupportedOperationException("Queries on replicated maps are not supported.");
+    }
+
+    @Override
+    public Map<String, LocalIndexStats> getIndexStats() {
+        throw new UnsupportedOperationException("Queries on replicated maps are not supported.");
+    }
+
+    @Override
+    public long getSetOperationCount() {
+        throw new UnsupportedOperationException("Set operation on replicated maps is not supported.");
+    }
+
+    @Override
+    public long getTotalSetLatency() {
+        throw new UnsupportedOperationException("Set operation on replicated maps is not supported.");
+    }
+
+    @Override
+    public long getMaxSetLatency() {
+        throw new UnsupportedOperationException("Set operation on replicated maps is not supported.");
+    }
+
+    @Override
+    public JsonObject toJson() {
+        return jsonObject;
+    }
+
+    @Override
+    public void fromJson(JsonObject json) {
+
```

Comment:
seems better to throw an unsupported-operation exception since we can't populate an object of this class with this `fromJson` method.

Reply:
It will break an ability to serialize/deserialize this object. The empty stats still should be transmittable. From the outer world this method works correctly anyway.

Reply:
Maybe i'm missing something, but as far as i see, having `fromJson` means you can change the state of this object but in reality, whatever you pass to this method, you can't change the state. I was thinking, this false assumption should be prevented by throwing exception. So after deserialize, an attempt to use `fromJson` can fail fast and inform user.

Reply:
Yes, you are right but I think more important here is do not break a convention: Object==deserialize(serialize(Object)) because this stats along as other stats are serialized/deserialized within MemberState.
It is done by constructing an object and filling all fields by fromJson(JsonObject) method. Adding the special logic there for this empty stats as well as throwing exception there is unwanted in my opinion. Please correct me if I am wrong.